### DISCUSSION TITLE
How to inject several instance of the same class?

### DISCUSSION THREAD

**Discussion by shaked-okami** [hadashiA/VContainer#538] [Q&A]
I have a game where i have several instances of a scriptable object.
Since those scriptable objects are all of the same type, how could i distinguish then when injecting them?
I remember that zenject used to have a with id parameter that offer that functionality, however i don't see it in VContainer.
I could just create empty classes that would inherit form the scriptable object so far, but i don't really like that solution.

**Comment by zadi0648** [comment]
If you have:

```
        builder.Register<IEnemy, AttackDog>(Lifetime.Singleton);
        builder.Register<IEnemy, AttackPig>(Lifetime.Singleton);
        builder.Register<IEnemy, AttackRat>(Lifetime.Singleton);
        builder.Register<IEnemy, AttackRatKing>(Lifetime.Singleton);
        builder.Register<IEnemy, DungBall>(Lifetime.Singleton);
```
        
you can do

```
[Inject]
private void InjectSomeStuff(IEnumerable<IEnemy> enemies)
```
    
or

`Resolver.Resolve(typeof(IEnumerable<IEnemy>)) as IEnumerable<IEnemy>;`

etc.
